src/main/java/nl/thehyve/ocdu/factories/PatientDataFactory.java

```diff
@@ -31,22 +37,50 @@ public PatientDataFactory(OcUser user, UploadSession submission) {
     }
 
     public List<Subject> createPatientData(Path patientFile) {
-//        try {
-//            Stream<String> lines = Files.lines(patientFile);
-//            Stream<String> lines2 = Files.lines(patientFile);
-//            Optional<String> headerLine = lines2.findFirst();
-//            HashMap<String, Integer> header = parseHeader(headerLine.get());
-//            HashMap<String, Integer> coreColumns = getCoreHeader(header);
-//            List<ClinicalData> clinicalData = new ArrayList<>();
-//            List<List<ClinicalData>> clinicalDataAggregates = lines.skip(1). // skip header
-//                    filter(s -> s.split(FILE_SEPARATOR).length > 2). // smallest legal file consists of no less than 3 columns
-//                    map(s -> parseLine(s, header, coreColumns)).collect(Collectors.toList());
-//            clinicalDataAggregates.forEach(aggregate -> clinicalData.addAll(aggregate));
-//            lines.close();
-//            lines2.close();
-//            return clinicalData;
-//        }
-
-        return null;
+        Optional<String[]> headerRow = getHeaderRow(patientFile);
+        if(headerRow.isPresent()) {
+            Map<String, Integer> columnsIndex = createColumnsIndexMap(headerRow.get());
+
+            try(Stream<String> lines = Files.lines(patientFile)) {
+                return lines.skip(1)
+                            .map(UserSubmittedDataFactory::parseLine)
+                            .map(row -> mapRow(row, columnsIndex))
+                            .collect(Collectors.toList());
+
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        else {
+            throw new RuntimeException("Patient file is empty.");
+        }
+
+    }
+
+    protected Subject mapRow(String[] row, Map<String, Integer> columnsIndex) {
```

Comment:
Why is it protected and not private? Do you expect any class to inherit from PatientDataFactory and use it?